### PR TITLE
feat(unlock-app): Support forever in recurring option

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

fixes https://github.com/unlock-protocol/unlock/issues/9867

Introduces a totalApproval field in the purchaseKeys which sets the total approval instead of using calculated total based on price and renewal times. 

CleanShot 2022-10-13 at 23 49 04@2x
CleanShot 2022-10-13 at 23 51 14@2x

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by kalidiagne** [comment]
need to add changelog and bump version for `unlock-js`

**Review by kalidiagne — APPROVED** [review]

**Comment by julien51** [comment]
Let's get @ChialiTsai 's feedback I'd design here please. I am not sure this yes the right approach.

File: unlock-app/src/components/interface/checkout/alpha/Checkout/Select.tsx [hunk 1]
```diff
@@ -254,37 +255,56 @@ export function Select({ checkoutService, injectedProvider }: Props) {
...
+                                      Renew{' '}
+                                      {typeof item.recurringPayments ===
+                                      'number'
+                                        ? `${item.recurringPayments} times`
+                                        : item.recurringPayments}
+                                    </Badge>
+                                  )}
+                                  {isMember && (
+                                    <Badge size="tiny" variant="primary">
+                                      Owned
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/alpha/Checkout/Select.tsx:272]: I think it is a great idea to show to the user when they already have a membership to a lock, but I am not sure this UI is the right one :/

File: unlock-app/src/components/interface/checkout/alpha/Checkout/Select.tsx [hunk 2]
```diff
@@ -254,37 +255,56 @@ export function Select({ checkoutService, injectedProvider }: Props) {
...
-                                      formattedData.isSoldOut
-                                        ? 'Sold out'
-                                        : formattedData.formattedKeysAvailable
-                                    }
-                                  />
+
+                              <div className="w-full space-y-2">
+                                <div className="flex items-center w-full grid-cols-2 gap-2">
+                                  {item.recurringPayments && (
+                                    <Badge variant="primary" size="tiny">
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/alpha/Checkout/Select.tsx:262]: I am not sure this is the right place to show this... and even if it is I don;t think this is the right UI.

I think confirmation is a much better place.

**Review by julien51 — CHANGES_REQUESTED** [review]
I would love @ChialiTsai 's comments on the best way to show these 2 elements:
* user already has a membership to the lock
* membership will renew.

**Comment by searchableguy** [comment]
As mentioned - this is what we have in the figma design. Happy to make the changes once I get new design.

**Comment by ChialiTsai** [comment]
image


May I propose the UI change to this? @searchableguy

**Comment by searchableguy** [comment]
@julien51 is above good to implement?

**Comment by julien51** [comment]
Yes! That's much better imo

**Comment by searchableguy** [comment]
CleanShot 2022-10-18 at 00 56 15@2x

**Review by julien51 — APPROVED** [review]

**Pull request merged**